src/Chapter3.hs

```diff
@@ -480,6 +528,89 @@ After defining the city, implement the following functions:
    and at least 10 living __people__ inside in all houses of the city totally.
 -}
 
+data Castle = Castle
+  {
+    name :: String
+  }
+
+data IsCastle = 
+  Yes Castle |
+  No
+
+data WhichBuilding = 
+  Chruch |
+  Library
+
+data People = 
+  One Int |
+  Two Int |
+  Three Int |
+  Four Int
```

Comment:
`Int` is not very suitable in here, as you can do something like `One 1000`, and it would mean that 1000 people could leave in One-People house 😸 

As you already have constructors separating the type of the house, you can omit having `Int` in there at all, and create an additional function instead, like:

```haskell
numberOfPeople :: People -> Int
```